pkg/plugin/connector/service.go

```diff
@@ -26,29 +26,41 @@ import (
 // registry is an object that can create new plugin dispensers. We need to use
 // an interface to prevent a cyclic dependency between the plugin package and
 // builtin and standalone packages.
-// There are two registries that implement this interface:
-//   - The built-in registry creates a dispenser which dispenses a plugin adapter
-//     that communicates with the plugin directly as if it was a library. These
-//     plugins are baked into the Conduit binary and included at compile time.
-//   - The standalone registry creates a dispenser which starts the plugin in a
-//     separate process and communicates with it via gRPC. These plugins are
-//     compiled independently of Conduit and can be included at runtime.
+// There are two registries that implement this interface: builtinReg and standaloneReg
 type registry interface {
-	NewDispenser(logger log.CtxLogger, name plugin.FullName) (Dispenser, error)
+	NewDispenser(logger log.CtxLogger, name plugin.FullName, connectorID string) (Dispenser, error)
 	List() map[plugin.FullName]pconnector.Specification
 }
 
+// The built-in registry creates a dispenser which dispenses a plugin adapter
+// that communicates with the plugin directly as if it was a library. These
+// plugins are baked into the Conduit binary and included at compile time.
+type builtinReg interface {
+	registry
+
+	Init(context.Context)
```

Comment:
Maybe we can make our lives easier and just add `Init(ctx context.Context, connUtilsAddr string, connUtilsToken string)` to `registry`, the builtin one can ignore the params it doesn't need 🤷

Reply:
I started with that, but given that the connector utils params are specific for the standalone, I went this route.